packages/ketcher-react/src/script/editor/Editor.ts

```diff
@@ -538,6 +557,289 @@ class Editor implements KetcherEditor {
+        ? 'end'
+        : 'begin';
+      const leavingAtomId = bond[bondEdgeForLeavingAtom];
+      const leavingAtom = currentStruct.atoms.get(leavingAtomId);
+      const newLeavingAtom = new Atom({ label: 'H', pp: leavingAtom?.pp });
+      const newLeavingAtomId = selectedStruct.atoms.add(newLeavingAtom);
+      this.atomIdsMap.set(newLeavingAtomId, leavingAtomId);
+
+      const bondEdgeForAttachmentAtom =
+        bondEdgeForLeavingAtom === 'end' ? 'begin' : 'end';
+      const attachmentAtomId = bond[bondEdgeForAttachmentAtom];
+      const attachmentAtomIdInSelectedStruct = atomIdsMap.get(attachmentAtomId);
+
+      if (attachmentAtomIdInSelectedStruct === undefined) {
+        return;
+      }
+
+      const newBond = new Bond({
+        type: Bond.PATTERN.TYPE.SINGLE,
+        begin: attachmentAtomIdInSelectedStruct,
+        end: newLeavingAtomId,
+      });
+      selectedStruct.bonds.add(newBond);
+
+      attachmentPoints.set(attachmentAtomIdInSelectedStruct, newLeavingAtomId);
+      this.atomIdsMap.set(attachmentAtomIdInSelectedStruct, attachmentAtomId);
+    });
+
+    this._monomerCreationState = {
+      originalStruct: currentStruct.clone(),
+      attachmentPoints,
+    };
+
+    this.render.monomerCreationRenderState = {
+      attachmentPoints,
+    };
+
+    this.struct(selectedStruct);
+  }
+
+  closeMonomerCreationWizard() {
+    if (!this._monomerCreationState) {
+      return;
+    }
+
+    this.render.monomerCreationRenderState = null;
+    this.struct(this._monomerCreationState.originalStruct);
+    this._monomerCreationState = null;
+
+    this.tool('select');
+  }
+
+  saveNewMonomer(data) {
+    if (!this._monomerCreationState) {
+      throw new Error(
+        'Monomer creation wizard is not active, cannot save new monomer',
+      );
+    }
+
+    const ketSerializer = new KetSerializer();
+    const ketMicromolecule = JSON.parse(
+      ketSerializer.serialize(this.render.ctab.molecule),
+    );
+
+    const { symbol, name, type, naturalAnalogue } = data;
+
+    const attachmentPoints: IKetAttachmentPoint[] = [];
+    this._monomerCreationState.attachmentPoints.forEach(
+      (leavingAtomId, attachmentAtomId) => {
+        const attachmentPoint: IKetAttachmentPoint = {
+          attachmentAtom: attachmentAtomId,
+          leavingGroup: {
+            atoms: [leavingAtomId],
+          },
+          type:
+            attachmentPoints.length === 0
+              ? 'left'
+              : attachmentPoints.length === 1
+              ? 'right'
+              : 'side',
+        };
+        attachmentPoints.push(attachmentPoint);
+      },
+    );
+
+    const monomerTemplate: IKetMonomerTemplate = {
+      type: KetTemplateType.MONOMER_TEMPLATE,
+      id: `${symbol}___${name}`,
+      class: type,
+      classHELM:
+        type === KetMonomerClass.AminoAcid
+          ? MONOMER_CONST.PEPTIDE
+          : type === KetMonomerClass.CHEM
+          ? MONOMER_CONST.CHEM
+          : MONOMER_CONST.RNA,
```

Comment:
Looks like there should be better place for this logic, because it is related to entities classes conversion